accepted/2023/vscode-commands.md

```diff
@@ -0,0 +1,221 @@
+
+At the Project level node in the solution explorer, we plan to add an item to the right-click menu that reads “Add a NuGet Package”.
+
+![Alt text](../../meta/resources/vscode-commands/addpackage-rightclick.png)
+
+Additionally, in the solution explorer, there is a "Dependencies" node, and within that node, there is a "Packages" node   , which contains a list of all the NuGet packages directly installed into the solution in question. This folder only is visible in the solution explorer if there are already previously installed NuGet packages in the project. If the Packages folder is visible, when a user hovers over it, they will see a small icon showing the NuGet logo with a plus sign next to it appear in line on the far right  (actual icon to be used: pending design).
+
+![Alt text](../../meta/resources/vscode-commands/addpackage-icon.png)
+
+Additionally, there will be another right-click menu on the Packages node  (if it is visible), providing them with options to manage their NuGet packages. Specifically, they will see an option to "Add package...".
+
+![Alt text](../../meta/resources/vscode-commands/addpackage-rightclick2.png)
+
+### Browse and Install a Package - Operation Flow
+
+First, the user will see a search box to provide a search term for a respective package to be found on their package sources. If the user wants to search specifically by a package attribute , and nuget.org is one of the configured package sources, they can do so by following the convention 'owner:Microsoft'. Otherwise, the search will search across all package attributes. (To see more information about what is supported with the syntax, see here : [Find and evaluate NuGet packages | Microsoft Learn](https://learn.microsoft.com/en-us/nuget/consume-packages/finding-and-choosing-packages))
+
+![Alt text](../../meta/resources/vscode-commands/addpackage-searchbox.png)
+
+The developer can then enter a search term to search for a NuGet package, and press enter. At this point, if they would like to cancel the operation, they can press the 'Escape' key on the keyboard. There will be a line of text underneath the search box that alerts the user of the functions of both the 'Enter' and 'Escape' keys here.
+
+When the developer presses enter, the results box will appear and provide them a selectable list of packages  that match their search term.  The search results will show the top 50 relevant packages which match this search term. For the first iteration of this experience, only the package ID will be shown in the search results. There are a few other possible options for what the search results could look like in the "Future Possibilities" section later in this document.
+
+![Alt text](../../meta/resources/vscode-commands/addpackage-searchbox2.png)
+
+Once a package is selected from the list, the user will then be prompted to select the version number that they would like to install.
+
+![Alt text](../../meta/resources/vscode-commands/addpackage-versionselect.png)
```

Comment:
Should this prefer the highest stable version? Imagine a user prefers to install stable versions but a package's 10 highest versions are all pre-release, this would be a frustrating experience.

Reply:
Great callout, yes this will depend on the settings a user provides similar to Visual Studio checking the include pre-release checkbox. It will prefer highest stable by default.